packages/mui-material/src/Checkbox/Checkbox.js

```diff
@@ -139,8 +142,22 @@ const Checkbox = React.forwardRef(function Checkbox(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const externalForwardedProps = {
+    ...other,
+    slots,
+    slotProps,
+  };
+
+  const [RootSlot, rootProps] = useSlot('root', {
+    elementType: CheckboxRoot,
+    ref,
+    className: clsx(classes.root, className),
+    ownerState,
+    externalForwardedProps,
+  });
+
   return (
-    <CheckboxRoot
+    <RootSlot
```

Comment:
Does it make sense to deprecate `inputProps` in favor of `slotProps.input`? cc @DiegoAndai